substrate/client/chain-spec/src/genesis_config_builder.rs

```diff
@@ -84,30 +84,46 @@ where
 	/// Returns a json representation of the default `RuntimeGenesisConfig` provided by the
 	/// `runtime`.
 	///
-	/// Calls [`GenesisBuilder::create_default_config`](sp_genesis_builder::GenesisBuilder::create_default_config) in the `runtime`.
+	/// Calls [`GenesisBuilder::get_preset`](sp_genesis_builder::GenesisBuilder::get_preset) in the
+	/// `runtime` with `None` argument.
 	pub fn get_default_config(&self) -> core::result::Result<Value, String> {
+		self.get_named_preset(None)
+	}
+
+	/// Returns a JSON blob representation of the builtin `GenesisConfig` identified by `id`.
+	///
+	/// Calls [`GenesisBuilder::get_preset`](sp_genesis_builder::GenesisBuilder::get_preset)
+	/// provided by the `runtime`.
+	pub fn get_named_preset(&self, id: Option<&String>) -> core::result::Result<Value, String> {
 		let mut t = BasicExternalities::new_empty();
 		let call_result = self
-			.call(&mut t, "GenesisBuilder_create_default_config", &[])
+			.call(&mut t, "GenesisBuilder_get_preset", &id.encode())
 			.map_err(|e| format!("wasm call error {e}"))?;
-		let default_config = Vec::<u8>::decode(&mut &call_result[..])
+
+		let named_preset = Option::<Vec<u8>>::decode(&mut &call_result[..])
 			.map_err(|e| format!("scale codec error: {e}"))?;
-		Ok(from_slice(&default_config[..]).expect("returned value is json. qed."))
+
+		if let Some(named_preset) = named_preset {
+			Ok(from_slice(&named_preset[..]).expect("returned value is json. qed."))
+		} else {
+			Err(format!("The preset with name {id:?} is not available."))
+		}
 	}
 
-	/// Builds `RuntimeGenesisConfig` from given json blob and returns the genesis state.
-	///
-	/// Calls [`GenesisBuilder::build_config`](sp_genesis_builder::GenesisBuilder::build_config)
-	/// provided by the `runtime`.
+	/// Calls [`sp_genesis_builder::GenesisBuilder::build_state`] provided by runtime.
 	pub fn get_storage_for_config(&self, config: Value) -> core::result::Result<Storage, String> {
 		let mut ext = BasicExternalities::new_empty();
 
+		let json_pretty_str = serde_json::to_string_pretty(&config)
+			.map_err(|e| format!("json to string failed: {e}"))?;
+
 		let call_result = self
-			.call(&mut ext, "GenesisBuilder_build_config", &config.to_string().encode())
+			.call(&mut ext, "GenesisBuilder_build_state", &json_pretty_str.encode())
 			.map_err(|e| format!("wasm call error {e}"))?;
 
 		BuildResult::decode(&mut &call_result[..])
-			.map_err(|e| format!("scale codec error: {e}"))??;
+			.map_err(|e| format!("scale codec error: {e}"))?
+			.map_err(|e| format!("{e} for blob:\n{}", json_pretty_str))?;
```

Comment:
This _pretty json_ ideally should be embedded into returned `Error` struct. For now it allows for better error finding by CLI user as @davxy pointed out during meeting.

Reply:
<details>
  <summary>Example</summary>
  
  ```
  Invalid JSON blob: BadLength at line 28 column 8 for blob:
  {
    "assignedSlots": {
      "config": null,
      "maxPermanentSlots": 0,
      "maxTemporarySlots": 0
    },
    "authorityDiscovery": {
      "keys": []
    },
    "babe": {
      "authorities": [],
      "epochConfig": {
        "allowed_slots": "PrimaryAndSecondaryVRFSlots",
        "c": [
          1,
          4
        ]
      }
    },
    "balances": {
      "balances": [
        [
          "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
          1000000000000000000
        ],
        [
          "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694tyx",
          1000000000000000000
        ],
  ```
</details>